jupyter_client/kernelspecapp.py

```diff
@@ -270,6 +271,22 @@ def start(self):
             self.exit(e)
 
 
+class ListProvisioners(JupyterApp):
+    version = __version__
+    description = """List available provisioners for use in kernel specifications."""
+
+    def start(self):
+        kfp = KernelProvisionerFactory(parent=self).instance()
```

Comment:
I believe this should be:
```
KernelProvisionerFactory.instance(parent=self)
```
Instantiating this class using the __init__ method doesn’t set a global instance of this singleton. When instance is called, it creates a new instance of this class and overrides the original one created. As a consequence, parent is not actually set.

Reply:
Good catch Zach - thank you.  I've updated this occurrence (the other two were correct).